modules/storage/storage-account/main.bicep

```diff
@@ -0,0 +1,68 @@
+@description('Deployment Location')
+param location string
+
+@description('Name of Storage Account. Must be unique within Azure.')
+param name string = 'st${uniqueString(resourceGroup().id, subscription().id)}'
+
+@allowed([ 'new', 'existing' ])
+@description('Specifies whether to create a new Storage Account or use an existing one.')
+param newOrExisting string = 'new'
+
+@description('Name of Resource Group where the Storage Account will be deployed.')
+param resourceGroupName string = resourceGroup().name
```

Comment:
this parameter is confusing me since it is only used for the `existing` resource - if I were to set this parameter to a different RG, it wouldn't deploy to that RG.

Reply:
good point, let me ponder this one....

Reply:
updated name to 'existingResourceGroupName', and updated description to example that it should be used for existing resources.